Strip CRLF from headers before parsing

m-czernek opened the pull request:
### What does this PR do?

After Tornado 6.4, 6.5 and later break compatibility in the `parse_response_start_line`, which no longer accepts CRLF. See [0] for more information.

This change ensures that fileclient tests (and functionality) do not break when you decide to upgrade Tornado :).

[0] https://github.com/tornadoweb/tornado/pull/3540

### What issues does this PR fix or reference?

### Previous Behavior
Fileclient download functionality (used e.g. in `cp.cache_file`) is broken with Tornado > 6.4.

### New Behavior
Fileclient download functionality works with Tornado > 6.4.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the test documentation for details on how to implement tests
into Salt's test suite:
https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

<!-- Please review Salt's Contributing Guide for best practices and guidance in
choosing the right branch:
https://docs.saltproject.io/en/master/topics/development/contributing.html -->

<!-- Additional guidance for pull requests can be found here:
https://docs.saltproject.io/en/master/topics/development/pull_requests.html -->

<!-- See GitHub's page on GPG signing for more information about signing commits
with GPG:
https://help.github.com/articles/signing-commits-using-gpg/ -->

m-czernek committed:
Strip CRLF from headers before parsing

twangboy reviewed (changes requested):
This will also need a test, please.

m-czernek commented:
@twangboy tests added as requested :+1:

m-czernek commented:
@twangboy I see that this is already fixed with https://github.com/saltstack/salt/commit/0501c71dd8ab7408b7ca0e2c820f34bcb39d0c90

Two problems I can see:

1. It's missing tests.
2. It uses `log.trace` which I couldn't find in the official docs - I've fixed it to `debug` in this PR.

I've rebased my PR, feel free to close it if you feel like the commit is enough, or accept it to fix the two problems I described. Up to y'all.
